src/backend/address_space_core.h

```diff
@@ -310,8 +310,11 @@ namespace snmalloc
       {
         if (rsize > size)
         {
-          add_range<PAL, Pagemap>(
-            local_state, pointer_offset(res, size), rsize - size);
+          size_t residual_size = rsize - size;
```

Comment:
Oh, blech, yes, that's a good point; thank you for catching it.  Probably that means that we shouldn't be bounding inside the pool of free objects and should just do so when we definitely reserve space, since we'll do so with a much smaller mantissa than CHERI supports...